src/coreclr/src/gc/gc.cpp

```diff
@@ -36643,26 +39172,26 @@ gc_heap::verify_heap (BOOL begin_gc_p)
                                         }
                                     }
                                 }
-        #endif //COLLECTIBLE_CLASS
+#endif //COLLECTIBLE_CLASS
 
                                 if (contain_pointers(curr_object))
                                 {
                                     go_through_object_nostart
                                         (method_table(curr_object), curr_object, s, oo,
                                         {
-                                            if ((crd != card_of ((uint8_t*)oo)) && !found_card_p)
+                                            if (crd != card_of ((uint8_t*)oo))
```

Comment:
I don't understand this change - looks like it's a bug fix?

Reply:
yes this was a bug fix. I noticed it when I was testing regions and I knew the cards were not correct yet the last GC's verify_heap didn't find it which was because we should always update found_card_p and need_card_p every time we see a new card instead of only updating them when found_card_p is FALSE.